src/system.adoc

```diff
@@ -0,0 +1,79 @@
+[appendix]
+== CHERI System Implications
+
+CHERI processors need memory systems which support the capability validity tags in memory.
+
+There are, or will soon be, a wide range of CHERI systems in existence from tiny IoT devices up to server chips.
+
+There are two types of bus connections used in chips which contain CHERI CPUs:
+
+. Tag-aware busses, where the bus protocol is extended to carry the tag along with the data.  This is typically done using a user defined bit in the protocol.
+.. These busses will read tags from memory (if tags are present in the target memory) and return them to the requestor.
+.. These busses will write the tag to memory as an extension of the data write.
+. Non-tag aware busses, i.e. normal non-CHERI aware busses.
+.. Reads of tagged memory will not read the tag.
+.. Writes to tagged memory will clear the tag of any CLEN-aligned CLEN-wide memory location where any byte matches the memory write.
+
+The fundamental rule for any CHERI system is that the tag and data are always accessed atomically. For every CLEN-aligned CLEN-wide memory location in memory It must never be possible to:
+
+. Update any data bytes without also writing the tag
+.. This implies clearing the tag if a non-CHERI aware bus master overwrites a capability in memory
+. Set the tag without also writing the data.
```

Comment:
```suggestion
. Read a tagged value with mismatched (stale or newer) data
. Set the tag without also writing the data.
```
This is implicit in the "accessed atomically" but might be worth calling out explicitly?